lib/hutch/main_loop.rb

```diff
@@ -0,0 +1,41 @@
+require 'hutch/logging'
+
+module Hutch
+  class MainLoop
```

Comment:
Most of MainLoop is inspired by this PR on sidekiq: https://github.com/mperham/sidekiq/pull/761

Not sure if loop is the correct name for this anymore, but I am struggling to think of a better one.